src/scores.jl

```diff
@@ -42,6 +42,15 @@ struct CompositeScore <: AbstractScore
 end
 
 
+struct ChainScoreData
+    """ The ChainScoreData object stores the values returned by score functions
+        at every step of the chain as well as the scores themselves.
+    """
+    scores::Array{S,1} where {S<:AbstractScore} # scores that were measured on a particular chain
+    step_values::Array{Dict{String, Any}} # array of Dicts which map {score name: score value}
+end
+
+
```

Comment:
Where the new class is defined! The `scores` array is helpful because storing the type of `AbstractScore` means that we can more efficiently reconstruct the score values in `get_score_values`, because we'll know if it's a district-level score or plan-level score, without needing to do hacky checks.

Reply:
great!!!! quick q: did you test that you can open a stored scores file from a script/notebook that is different than the one the chain was run on? I want to make sure that you can retrieve the AbstractScores perfectly.

Otherwise this is a great addition!!!

Reply:
Good q! Just tested this using the `Serialization` library and everything worked without a hitch 👍